ui/helpers/utils/util.js

```diff
@@ -715,6 +716,29 @@ export const isAbleToExportAccount = (keyringType = '') => {
   return !keyringType.includes('Hardware') && !keyringType.includes('Snap');
 };
 
+export const isAbleToRevealSrp = (accountToExport, keyrings) => {
+  const {
+    metadata: {
+      keyring: { type },
+    },
+  } = accountToExport;
+  if (type !== KeyringTypes.hd && type !== KeyringTypes.snap) {
+    return false;
+  }
+
+  // All hd keyrings can reveal their srp.
+  if (type === KeyringTypes.hd) {
+    return true;
+  }
+
+  // Not all snaps have an entropy source
+  const keyringId = accountToExport.options?.entropySource;
+  return keyrings.some(
+    (keyring) =>
+      keyring.type === KeyringTypes.hd && keyring.metadata.id === keyringId,
+  );
+};
```

Comment:
WDYT about only using 1st party Snaps here too?

```suggestion
  const {
    metadata: {
      keyring: { type },
      snap,
    },
    options: {
      entropySource,
    },
  } = accountToExport;

  // All hd keyrings can reveal their srp.
  if (type === KeyringTypes.hd) {
    return true;
  }

  // We only consider 1st-party Snaps that have an entropy source.
  if (type === KeyringTypes.snap && isMultichainWalletSnap(snap.id) && entropySource) {
    const keyringId = entropySource;
    return keyrings.some(
      (keyring) =>
        keyring.type === KeyringTypes.hd && keyring.metadata.id === keyringId,
    );
  }
  
  return false;
};
```